packages/components/inputs/checkbox-input/src/checkbox-input.tsx

```diff
@@ -247,10 +239,32 @@ const CheckboxIcon = (props: TLabelProps) => {
           return null;
         })()}
       </div>
-    </div>
+    </CheckboxIconWrapper>
   );
 };
 
+const Label = styled.label<TLabelProps>`
+  display: flex;
+  align-items: center;
+  cursor: ${(props) => {
+    if (props.isDisabled) return 'not-allowed';
+    if (props.isReadOnly) return 'default';
+    return 'pointer';
+  }};
+  position: relative;
+
+  &:focus-within ${LabelTextWrapper} {
+    box-shadow: 0 0 0 2px ${designTokens.borderColorForInputWhenFocused};
+  }
+
+  &:hover ${CheckboxIconWrapper} {
+    background-color: ${(props) =>
+      props.isDisabled || props.isReadOnly
+        ? 'unset'
+        : designTokens.backgroundColorForCheckboxInputIconWhenHovered};
+  }
```

Comment:
Move the hover effect into the `Label` component. Note that we are targeting the `CheckboxIconWrapper` using the [Emotion feature](https://emotion.sh/docs/styled#targeting-another-emotion-component).